Samples/ios/SampleAppiOS-Swift/SampleAppiOS-Swift/SampleMainViewController.swift

```diff
@@ -0,0 +1,179 @@
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+import UIKit
+
+class SampleMainViewController: UIViewController {
+
+    @IBOutlet var profileImageView: UIImageView!
+    @IBOutlet var nameLabel: UILabel!
+    @IBOutlet var tableView: UITableView!
+    
+    var events: [Date: [SampleCalendarEvent]]!
+    var keys: [Date]!
+    
+    fileprivate static let s_dayFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateStyle = .full
+        formatter.timeStyle = .none
+        return formatter
+    }()
+    
+    fileprivate static let s_timeFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateStyle = .none
+        formatter.timeStyle = .short
+        return formatter
+    }()
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
+        tableView.dataSource = self
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        
+        loadPhoto()
+        loadEvents()
+        
+        do {
+            let user = try SampleMSALUtil.shared.currentUser()
+            nameLabel.text = "Welcome, \(user.name!)"
+        } catch let error {
+            print("Loading current user name error: \(error)")
+        }
+        
+        
+    }
+    
+    @IBAction func signOut(_ sender: Any) {
+        do {
+            try SampleMSALUtil.shared.signOut()
+        } catch let error {
+            print("Sign out error: \(error)")
+        }
+        
+        let appDelegate = UIApplication.shared.delegate as! AppDelegate
+        appDelegate.showLoginVC()
+    }
+
+}
+
+// MARK: Loading/setting photo and events
+fileprivate extension SampleMainViewController {
+    
+    func loadPhoto() {
+        let photoUtil = SamplePhotoUtil.shared
+        setUserPhoto(withPhoto: photoUtil.cachedPhoto())
+        
+        photoUtil.checkUpdatePhoto {
+            (image, error) in
+            
+            if let error = error {
+                print("checkUpdatePhoto error: \(error)")
```

Comment:
nit: shouldn't we return here? Theoretically if error != nil, image == nil anyway, but just in case...